### PR TITLE
Scaffolding for Shapefile module

### PR DESCRIPTION
**Pull request by kylebarron**
Adds test files from `mbostock/shapefile` and other general module files. Intended for #800 to have a cleaner diff.

### CODE REVIEW

**Comment by kylebarron** [comment]
I'm not sure exactly what files to include in this PR so that CI passes... delete everything but the test files?

File: modules/shapefile/docs/api-reference/shp-loader.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const data = await load(url, SHPLoader);
+```
+
+## Options
+
+N/A
+
+## Format Summary
+
+Well-known binary (SHP) is a binary geometry encoding to store geometries (it
```
Comment (ibgreen) [inline modules/shapefile/docs/api-reference/shp-loader.md:48]: Is it WKB or not? maybe be a little more precise? Assume this section was copied and needs an update.

File: modules/shapefile/test/data/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Data Attribution
+
+- Files in the `bostock/` folder are from `mbostock/shapefile`
```
Comment (ibgreen) [inline modules/shapefile/test/data/README.md:3]: Specify what license (add it to the folder). Also his license forbids the use of his name, maybe pick another directory name :)

File: modules/shapefile/docs/api-reference/shp-loader.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  <img src="https://img.shields.io/badge/From-v2.2-blue.svg?style=flat-square" alt="From-v2.2" />
+</p>
+
+Shapefile loader
+
+| Loader                | Characteristic                                |
+| --------------------- | --------------------------------------------- |
+| File Extension        | `.shp`,                                       |
+| File Type             | Binary                                        |
+| File Format           | [Well Known Binary][wkb]                      |
```
Comment (ibgreen) [inline modules/shapefile/docs/api-reference/shp-loader.md:13]: Update?

File: modules/shapefile/docs/api-reference/shp-loader.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+| Data Format           | [Geometry](/docs/specifications/category-gis) |
+| Supported APIs        | `load`, `parse`, `parseSync`                  |
+| Decoder Type          | Synchronous                                   |
+| Worker Thread Support | Yes                                           |
+
+## Usage
+
+```js
+import {SHPLoader} from '@loaders.gl/wkt';
```
Comment (ibgreen) [inline modules/shapefile/docs/api-reference/shp-loader.md:22]: module name

File: modules/shapefile/docs/api-reference/shp-loader.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  1, 1, 0, 0,   0,  0,  0,
+  0, 0, 0, 0, 240, 63,  0,
+  0, 0, 0, 0,   0,  0, 64
+]).buffer;
+const data = parseSync(buffer, SHPLoader);
+// => { positions: { value: Float64Array(2) [ 1, 2 ], size: 2 } }
+```
+
+```js
+import {SHPLoader} from '@loaders.gl/wkt';
```
Comment (ibgreen) [inline modules/shapefile/docs/api-reference/shp-loader.md:36]: module name

File: modules/shapefile/package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  "sideEffects": false,
+  "files": [
+    "src",
+    "dist",
+    "README.md"
+  ],
+  "scripts": {
+    "pre-build": "npm run build-worker && npm run build-bundle && npm run build-bundle -- --env.dev",
+    "build-bundle": "webpack --display=minimal --config ../../scripts/bundle.config.js",
+    "build-worker": "webpack --entry ./src/wkt-loader.worker.js --output ./dist/wkt-loader.worker.js --config ../../scripts/worker-webpack-config.js"
```
Comment (ibgreen) [inline modules/shapefile/package.json:32]: Update scripts

File: modules/shapefile/package.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "devDependencies": {
+    "fuzzer": "^0.2.1"
```
Comment (ibgreen) [inline modules/shapefile/package.json:35]: Can probably drop this one

File: modules/shapefile/docs/api-reference/shp-loader.md (hunk 1 above)
Comment (kylebarron) [reply modules/shapefile/docs/api-reference/shp-loader.md:48]: Yes it needs an update.. I'll modify before landing

**Pull request merged**